File: Android/WatchVerse/app/src/main/java/com/maximeproulx/watchverse/Universe.kt
package com.maximeproulx.watchverse

data class Universe(
    val id: String,
    val state: String = "available",
    val title: String,
    val subtitle: String,
    val fullTitle: String,
    val description: String,
    val logo: String = "",
    val banner: String,
    val poster: String,
    val filters: List<String> = emptyList(),
    val sortOrder: Int = 0,
    val movies: List<Movie> = emptyList()
)

File: IOS/Shared/Models/Universe.swift
//
//  Universe.swift
//  WatchVerse
//
//  Created by Maxime Proulx on 2026-07-30.
//

import Foundation

struct Universe: Identifiable, Codable {
    let id: String
    
    let title: String
    let subtitle: String
    let fullTitle: String
    
    let description: String
    
    let banner: String
    let poster: String
    
    let filters: [String]?
    
    let movies: [Movie]
    
    static let loading = Universe(
        id: "loading",
        title: "",
        subtitle: "",
        fullTitle: "",
        description: "",
        banner: "",
        poster: "",
        filters: nil,
        movies: []
    )
}
